intellij-toml/schema/src/main/kotlin/org/toml/ide/schema/TomlPropertyAdapter.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.toml.ide.schema
+
+import com.intellij.psi.PsiElement
+import com.jetbrains.jsonSchema.extension.adapters.JsonObjectValueAdapter
+import com.jetbrains.jsonSchema.extension.adapters.JsonPropertyAdapter
+import com.jetbrains.jsonSchema.extension.adapters.JsonValueAdapter
+import org.toml.lang.psi.TomlArray
+import org.toml.lang.psi.TomlElement
+import org.toml.lang.psi.TomlKeyValue
+import org.toml.lang.psi.TomlKeyValueOwner
+
+class TomlPropertyAdapter(private val tomlKeyValue: TomlKeyValue) : JsonPropertyAdapter {
```

Comment:
Maybe it's worth creation some base (sealed?) class like 

```kotlin
sealed class TomlPropertyAdapterBase<T : TomlElement>(protected val element: T) : JsonPropertyAdapter {
    override fun getDelegate(): PsiElement = element

    companion object {
        fun createAdapterByType(value: TomlElement): TomlPropertyAdapterBase<*> = when (value) {
            is TomlKeyValueOwner -> TomlObjectAdapter(value)
            is TomlArray -> TomlArrayAdapter(value)
            else -> TomlGenericValueAdapter(value)
        }
    }
}
```

and inherit all other toml adapters from it?